test/README.md

```diff
@@ -0,0 +1,88 @@
+# Test environments
+
+This repository's tests and development automation tasks are organized using [tox], a command-line CI frontend for Python projects.  tox is typically used during local development and is also invoked from this repository's GitHub Actions [workflows](../.github/workflows/).
+
+tox can be installed by running `pip install tox`.
+
+tox is organized around various "environments," each of which is described below.  To run _all_ test environments, run `tox` without any arguments:
+
+```sh
+$ tox
+```
+
+Environments for this repository are configured in [`tox.ini`] as described below.
+
+## Lint environment
+
+The `lint` environment ensures that the code meets basic coding standards, including
+
+- [_Black_] formatting style
+- Style checking with [ruff], [autoflake], and [pydocstyle]
+- [mypy] type annotation checker, as configured by [`.mypy.ini`]
+
+The _Black_ and mypy passes are applied also to [Jupyter] notebooks (via [nbqa]).
+
+To run:
+
+```sh
+$ tox -e lint
+```
+
+## Style environment
+
+The command `tox -e style` will apply automated style fixes.  This includes:
+
+- Automated fixes from [ruff] and [autoflake]
+- Reformatting of all files in the repository according to _Black_ style
+
+## Test (py##) environments
+
+The `py##` environments are the main test environments.  tox defines one for each version of Python.  For instance, the following command will run the tests on Python 3.8, Python 3.9, and Python 3.10:
+
+```sh
+$ tox -e py38,py39,py310
+```
+
+These environments execute all tests using [pytest], which supports its own simple style of tests, in addition to [unittest]-style tests and [doctests] located throughout the project's docstrings.
+
+## Notebook environments
+
+The `notebook` and `py##-notebook` environments invoke [nbmake] to ensure that all Jupyter notebooks in the [`docs/`](/docs/) directory execute successfully.
+
+## Coverage environment
+
+The `coverage` environment uses [Coverage.py] to ensure that the fraction of code tested by pytest is above some threshold (currently set to 80% overall, and 100% in all new modules).  A detailed, line-by-line coverage report can be viewed by navigating to `htmlcov/index.html` in a web browser.
```

Comment:
I notice Coverage.py and Jupyter are capitalized in the brackets, but many other packages are not. Also [_Black_] has the underscores. Are there any principles to capitalization for these packages?

Capitalization aside, do we want to include the file extension for coverage? We don't do that elsewhere here

Reply:
I followed the way each package's name is formatted.  For instance, https://coverage.readthedocs.io/en/7.2.7/ says "Coverage.py", presumably to disambiguate from other coverage tools.

Reply:
OK cool, I figured you followed some principle. Looks great, thanks